Mathlib/NumberTheory/Rayleigh.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2023 Jason Yuen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jason Yuen
+-/
+import Mathlib.Data.Real.ConjugateExponents
+import Mathlib.Data.Real.Irrational
+
+/-!
+# Rayleigh's theorem on Beatty sequences
+
+This file proves Rayleigh's theorem on Beatty sequences.
+
+## Main statements
+
+* `beattySequence`: The Beatty sequence for real number `r` is defined to be
+  `B_r := {⌊r⌋, ⌊2r⌋, ⌊3r⌋, ...}`.
+* `rayleigh`: Rayleigh's theorem on Beatty sequences. Let `r` be an irrational real number greater
+  than 1. Then every positive integer is in exactly one of `B_r` or `B_(r / (r - 1))`.
+* `rayleigh'`: Rayleigh's theorem on Beatty sequences. Let `r` and `s` be irrational real numbers
+  greater than 1, and satisfy `1/p + 1/q = 1`. Then every positive integer is in exactly one of
+  `B_r` or `B_s`.
+
+## References
+
+* [Wikipedia, *Beatty sequence*](https://en.wikipedia.org/wiki/Beatty_sequence)
+
+## Tags
+
+beatty, sequence, rayleigh, irrational, floor
+-/
+
+/-- The Beatty sequence for real number `r` is defined to be `B_r := {⌊r⌋, ⌊2r⌋, ⌊3r⌋, ...}`. -/
+def beattySequence (r : ℝ) : Set ℤ :=
```

Comment:
I wonder if it might be better to define this as an actual infinite sequence indexed by integers (i.e. a function from integers to integers) rather than a set. That seems useful for other applications of this sequence at least

Reply:
Done. I split the definition into `beattySequence` and `beattySet`.